### ISSUE TITLE
[Question] Shouldn't pred_pad be probabilities before applying argmax ?

### ISSUE THREAD

**Issue by pyyush** [espnet/espnet#2947]
# e2e_st_transformer.py
line 225 - 237
- criterion (LabelSmoothingLoss) applies log_softmax to pred_pad inside but th_accuracy and error_calculator don't and directly apply argmax!
 `loss_att = self.criterion(pred_pad, ys_out_pad)`
 `self.acc = th_accuracy(pred_pad.view(-1, self.odim), ys_out_pad, ignore_label=self.ignore_id)`
 `ys_hat = pred_pad.argmax(dim=-1)`
  `self.bleu = self.error_calculator(ys_hat.cpu(), ys_pad.cpu())`

**Comment by sw005320** [comment]
@hirofumi0810, could I ask you to comment on this question?

**Comment by hirofumi0810** [comment]
@pyyush This is not a speech translation specific question.
First of all, the current implementation is correct.
When calculating accuracy and task specific error rate (e.g., WER, BLEU), we pick up the best token id at each output timestep.
The score order in logits is the same as (log) probabilities.
Therefore, `log_softmax` operation is not necessary.
This is true when using greedy decoding as well.

**Comment by pyyush** [comment]
@hirofumi0810 Thanks for your explanation.
